packages/rest/test/unit/coercion/utils.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  OperationObject,
+  ParameterObject,
+  SchemaObject,
+} from '@loopback/openapi-v3-types';
+
+import {
+  ShotRequestOptions,
+  expect,
+  stubExpressContext,
+} from '@loopback/testlab';
+
+import {
+  PathParameterValues,
+  Request,
+  Route,
+  createResolvedRoute,
+  parseOperationArgs,
+  ResolvedRoute,
+} from '../../..';
+import * as HttpErrors from 'http-errors';
+
+export function givenOperationWithParameters(params?: ParameterObject[]) {
+  return <OperationObject>{
+    'x-operation-name': 'testOp',
+    parameters: params,
+    responses: {},
+  };
+}
+
+export function givenRequest(options?: ShotRequestOptions): Request {
+  return stubExpressContext(options).request;
+}
+
+export interface TestArgs<T> {
+  expectedResult: T;
+  valueFromReq: string;
+  schema?: SchemaObject;
+  specConfig?: Partial<ParameterObject>;
+  caller: string;
+  expectError: boolean;
+}
+
+export function givenResolvedRoute(
+  spec: OperationObject,
+  pathParams: PathParameterValues = {},
+): ResolvedRoute {
+  const route = new Route('get', '/', spec, () => {});
+  return createResolvedRoute(route, pathParams);
+}
+
+export async function testCoercion<T>(config: TestArgs<T>) {
+  try {
+    const req = givenRequest();
+    const spec = givenOperationWithParameters([
+      {
+        name: 'aparameter',
+        in: 'path',
+        schema: config.schema,
+      },
+    ]);
+    const route = givenResolvedRoute(spec, {aparameter: config.valueFromReq});
+
+    if (config.expectError) {
+      try {
+        await parseOperationArgs(req, route);
+        throw new Error("'parseOperationArgs' should throw error!");
+      } catch (err) {
+        expect(err).to.eql(config.expectedResult);
+      }
+    } else {
+      const args = await parseOperationArgs(req, route);
+      expect(args).to.eql([config.expectedResult]);
+    }
+  } catch (err) {
+    err.stack += config.caller;
+    throw err;
+  }
+}
+
+// tslint:disable-next-line:no-any
+export function test(t: any[]) {
```

Comment:
A screenshot of the error report:
![screen shot 2018-06-06 at 3 05 11 pm](https://user-images.githubusercontent.com/12554153/41059679-3557fb74-699b-11e8-8013-7ce46b1eb8e0.png)
It contains the caller's information in stack.

Reply:
> Let's use properly typed arguments instead of any[] array please.

Here is what I mean:

```ts
function test<T>(spec: SchemaObject, input: string, expectedOutput: T);
```

If you think there are test cases where a hand-written test name is better than the auto-generated one (`converts ${input} to ${expectedOutput}`), then I am proposing to add a new optional parameter for test case name at the end of the arguments list.

```ts
function test<T>(spec: SchemaObject, input: string, expectedOutput: T, testName?: string);
```

Thoughts?